test/unittests/libp2p/net.cpp

```diff
@@ -1173,6 +1147,128 @@ BOOST_AUTO_TEST_CASE(addNodePingsNodeOnlyOnce)
     BOOST_REQUIRE_EQUAL(sentPing->pingHash, sentPing2->pingHash);
 }
 
+class PacketsWithChangedEndpointFixture : public TestOutputHelperFixture
+{
+public:
+    PacketsWithChangedEndpointFixture()
+    {
+        nodeTableHost.start();
+        nodeSocketHost1.start();
+        nodePort1 = nodeSocketHost1.port;
+        nodeSocketHost2.start();
+        nodePort2 = nodeSocketHost2.port;
+
+        nodeEndpoint1 =
+            NodeIPEndpoint{bi::address::from_string(c_localhostIp), nodePort1, nodePort1};
+        nodeEndpoint2 =
+            NodeIPEndpoint{bi::address::from_string(c_localhostIp), nodePort2, nodePort2};
+
+        // add a node to node table, initiating PING
+        nodeTable->addNode(Node{nodePubKey, nodeEndpoint1});
+
+        // handle received PING
+        auto pingDataReceived = nodeSocketHost1.packetsReceived.pop();
+        auto pingDatagram =
+            DiscoveryDatagram::interpretUDP(bi::udp::endpoint{}, dev::ref(pingDataReceived));
+        auto ping = dynamic_cast<PingNode const&>(*pingDatagram);
```

Comment:
Can we validate the name of the ping packet here via BOOST_REQUIRE or is that not something we can do in a test fixture class?

Reply:
I think we can.